File: wrappers/csharp/Intel.RealSense/Types/Enums/ThreadCategory.cs
// License: Apache 2.0. See LICENSE file in root directory.
// Copyright(c) 2026 Intel Corporation. All Rights Reserved.

namespace Intel.RealSense
{
    /// <summary>
    /// Category of a library-created thread, for use with the thread-start callback
    /// </summary>
    public enum ThreadCategory
    {
        /// <summary> USB I/O threads </summary>
        UsbIo = 0,

        /// <summary> Video capture threads </summary>
        VideoCapture = 1,

        /// <summary> Sensor I/O threads </summary>
        SensorIo = 2,

        /// <summary> Frame processing threads </summary>
        FrameProcessing = 3,

        /// <summary> Device monitoring threads </summary>
        DeviceMonitoring = 4,

        /// <summary> Dispatch threads </summary>
        Dispatch = 5,

        /// <summary> Network threads </summary>
        Network = 6,

        /// <summary> Utility threads </summary>
        Utility = 7,
    }
}

File: wrappers/android/librealsense/src/main/java/com/intel/realsense/librealsense/ThreadCategory.java
package com.intel.realsense.librealsense;

public enum ThreadCategory {
    USB_IO(0),
    VIDEO_CAPTURE(1),
    SENSOR_IO(2),
    FRAME_PROCESSING(3),
    DEVICE_MONITORING(4),
    DISPATCH(5),
    NETWORK(6),
    UTILITY(7);

    private final int mValue;

    private ThreadCategory(int value) { mValue = value; }
    public int value() { return mValue; }
}
